golang/vaultclient/createAccount.go

```diff
@@ -0,0 +1,127 @@
+	Name     *string `locationName:"name"`
+	Email    *string `locationName:"emailAddress"`
+	QuotaMax *int64  `locationName:"quotaMax"`
+}
+
+// String returns the string representation
+func (s CreateAccountInput) String() string {
+	return awsutil.Prettify(s)
+}
+
+// Validate inspects the fields of the type to determine if they are valid.
+func (s *CreateAccountInput) Validate() error {
+	invalidParams := request.ErrInvalidParams{Context: "CreateAccountInput"}
+
+	if s.Name == nil {
+		invalidParams.Add(request.NewErrParamRequired("Name"))
+	}
+	if s.Name != nil && len(*s.Name) < 1 {
+		invalidParams.Add(request.NewErrParamMinLen("Name", 1))
+	}
+
+	if s.Email == nil {
+		invalidParams.Add(request.NewErrParamRequired("Email"))
+	}
+	if s.Email != nil && len(*s.Email) < 1 {
+		invalidParams.Add(request.NewErrParamMinLen("Email", 1))
+	}
+
+	if s.QuotaMax != nil && *s.QuotaMax < 1 {
+		invalidParams.Add(request.NewErrParamMinValue("QuotaMax", 1))
+	}
+
+	if invalidParams.Len() > 0 {
+		return invalidParams
+	}
+	return nil
+}
+
+// SetName sets the Name field's value.
+func (s *CreateAccountInput) SetName(v string) *CreateAccountInput {
+	s.Name = &v
+	return s
+}
+
+// SetEmail sets the Email field's value.
+func (s *CreateAccountInput) SetEmail(v string) *CreateAccountInput {
+	s.Email = &v
+	return s
+}
+
+// SetQuotaMax sets the QuotaMax field's value.
+func (s *CreateAccountInput) SetQuotaMax(v int64) *CreateAccountInput {
+	s.QuotaMax = &v
+	return s
+}
+
+// CreateAccount API operation creates a new Vault account
+func (c *Vault) CreateAccount(input *CreateAccountInput) (*CreateAccountOutput, error) {
+	req, out := c.CreateAccountRequest(input)
+	return out, req.Send()
+}
+
+// CreateAccountRequest generates a "aws/request.Request" representing the
+// client's request for the CreateAccount operation. The "output" return
+// value will be populated with the request's response once the request completes
+// successfully.
+//
+// Use "Send" method on the returned Request to send the API call to the service.
+// the "output" return value is not valid until after Send returns without error.
+func (c *Vault) CreateAccountRequest(input *CreateAccountInput) (req *request.Request, output *CreateAccountOutput) {
+	op := &request.Operation{
+		Name:       opCreateAccount,
+		HTTPMethod: "POST",
+		HTTPPath:   "/",
+	}
+
+	if input == nil {
+		input = &CreateAccountInput{}
+	}
+
+	output = &CreateAccountOutput{}
+	req = c.newRequest(op, input, output)
+	return
+}
+
+// Account contains information about a Vault account.
+type Account struct {
+	AccountData *AccountData `type:"structure" locationName:"data"`
+}
+
+type AccountData struct {
+	Arn         *string    `locationName:"arn"`
+	Name        *string    `locationName:"name"`
+	Email       *string    `locationName:"emailAddress"`
+	ID          *string    `locationName:"id"`
+	QuotaMax    *int64     `locationName:"quotaMax"`
+	CreateDate  *time.Time `locationName:"createDate"`
```

Comment:
Is create date also part of the response you get back?

Reply:
Yes, https://github.com/scality/Vault/blob/development/8.2/lib/Handlers/Account.js#L393

Reply:
yeah! you are right, I was confused whether we send that info back to the client.